doc_src/history.txt

```diff
@@ -36,12 +37,18 @@ These flags can appear before or immediately after one of the sub-commands liste
 
 - `-e` or `--exact` searches or deletes items in the history that exactly match the specified text string. This is the default for the `--delete` flag. Note that the match is case-insensitive by default. If you really want an exact match, including letter case, you must use the `-C` or `--case-sensitive` flag.
 
+- `--file` changes the current history file. That option can be used both independent of any other option and in conjunction with the other options. In both cases the change to the current history file is persistent and affects subsequent commands. The switch to a new history file occurs before any other option is acted on and an implicit `history --save` occurs before the switch.
+
+The "history_file_identifier" can be a simple word or a relative or absolute path name. A simple word (i.e., one without slashes) is interpreted as a file name relative to `$XDG_DATA_HOME/fish` (`$HOME/.local/share/fish` by default). If "history_file_identifier" is the special word "fish" the default history path name will be substituted. The special word "none" results in no history file being used. This is analogous to the incognito mode provided by web browsers like Chrome. A relative path (one that contains a slash but does not begin with a slash) is interpreted relative to `$PWD` at the time the command is run. The resulting path must refer to a regular file (or not exist). Special file names like /dev/null are not allowed.
+
 - `-p` or `--prefix` searches or deletes items in the history that begin with the specified text string. This is not currently supported by the `--delete` flag.
 
 - `-t` or `--show-time` prepends each history entry with the date and time the entry was recorded . By default it uses the strftime format `# %c%n`. You can specify another format; e.g., `--show-time='%Y-%m-%d %H:%M:%S '` or `--show-time='%a%I%p'`. The short option, `-t` doesn't accept a stftime format string; it only uses the default format. Any strftime format is allowed, including `%s` to get the raw UNIX seconds since the epoch. Note that `--with-time` is also allowed but is deprecated and will be removed at a future date.
 
 - `-z` or `--null` causes history entries written by the search operations to be terminated by a NUL character rather than a newline. This allows the output to be processed by `read -z` to correctly handle multiline history entries.
 
+If `--search` is specified without `--contains` or `--prefix`, `--contains` will be assumed.
```

Comment:
It is probably because @tomassedovic used my very moldy work-in-progress change as their starting point and incorrectly resolved some merge conflicts. My original change does not have that text but it does appear in subsequent commits.